modules/internal/String.chpl

```diff
@@ -1021,8 +1254,8 @@ module String {
                 string, or 0 if the `needle` is not in the string.
      */
     // TODO: better name than region?
-    proc find(needle: string, region: range(?) = 1..) : int {
-      return _search_helper(needle, region, count=false);
+    proc find(needle: string, region: range(?) = 1:byteIndex..) : byteIndex {
```

Comment:
Is there supposed to be a codepoint indexes or ints version of this with the region always specified? (To avoid ambiguity, don't make `region` optional in that one)

Reply:
This version of `string.find()` covers all three types.  The differentiation is made two call levels down in `_getView()`.

Reply:
Oh, I was mis-remembering; the `range(?)` makes the argument generic and not inferred from the default. Sounds good.